scene/gui/text_edit.cpp

```diff
@@ -2588,117 +2588,101 @@ void TextEdit::_do_backspace(bool p_word, bool p_all_to_left) {
 	Vector<int> caret_edit_order = get_caret_index_edit_order();
 	for (int i = 0; i < caret_edit_order.size(); i++) {
 		int caret_idx = caret_edit_order[i];
-		if (get_caret_column(caret_idx) == 0 && get_caret_line(caret_idx) == 0 && !has_selection(caret_idx)) {
+		if (has_selection(caret_idx)) {
+			delete_selection(caret_idx);
 			continue;
 		}
 
-		if (has_selection(caret_idx) || (!p_all_to_left && !p_word) || get_caret_column(caret_idx) == 0) {
-			backspace(caret_idx);
```

Comment:
This has to call `backspace` for a backspace as this allows inherited classes i.e `CodeEdit` to override the behaviour.

Reply:
Not sure why overriding `backspace()` would be desirable.

- Why only do this for BS and not Del for example?
- It only works for this specific case of "either a selection, in the 0th column, or default BS", wouldn't that be very weird even for those who do want to use it?

So with that said:
- I think the other `backspace()` function can't be smacked here in the middle unless I duplicate a lot of logic.
- Can I perhaps make this the official overrideable `backspace()` function? The other one is used only 3 times in our codebase, and this can replace its functionality if used without arguments.

Reply:
`CodeEdit` removes any brace pairs on backspace but not on delete, which is why it's overrideable and delete is not. So far, there hasn't been a need to provide a way to override delete.  Regardless, we should be consistent between the user pressing the backspace key and calling backspace through the API in the standard case.

I'm not sure what the best solution looks like, without doing some experimenting, but it wouldn't surprise me that some logic need to be duplicated.

Reply:
Ah yeah, this breaks the functionality. I'll think of something...

Reply:
Should be good now.